applybn/feature_extraction/bn_feature_extractor.py

```diff
@@ -1,312 +1,323 @@
         """
-        Initializes the BNFeatureGenerator.
+        Fits the BNFeatureGenerator to the data.
+
+        This involves:
+        1.  Adding the target variable (if provided) to the input data.
+        2.  Encoding categorical columns.
+        3.  Discretizing continuous columns.
+        4.  Creating a Bayesian Network based on the data types.
+        5.  Learning the structure of the Bayesian Network (if known_structure is not provided).
+        6.  Fitting the parameters of the Bayesian Network.
 
         Args:
-            known_structure (Optional[List[Tuple[str, str]]]): A list of edges
-                representing the known structure of the Bayesian Network.
+            X (pd.DataFrame): The input data.
+            y (Optional[pd.Series]): The target variable. If provided, it will be added to the input data and
+                treated as a node in the Bayesian Network.
+
+        Returns:
+            self: The fitted BNFeatureGenerator object.
         """
-        self.known_structure = known_structure
-        self.bn: Optional[BayesianNetwork] = None
-        self.variables: Optional[List[str]] = None
-        self.num_classes: Optional[int] = None
-        self.random_seed = random_seed
+        if y is not(None):
+            X = pd.concat([X, y], axis = 1).reset_index(drop=True)
+            self.target_name = y.name
+        cat_columns = X.select_dtypes(include=["object", "category"]).columns.tolist()
+
+        if cat_columns:
+            X, cat_encoder = code_categories(X, method="label", columns=cat_columns)
+        X = pd.DataFrame(X)
+        discretized_data, est = self.preprocessor.apply(X)
+        discretized_data = pd.DataFrame(discretized_data, columns=X.columns)
+        info = self.preprocessor.info # Get information about the data types after preprocessing
+        disc_columns = 0
+        cont_columns = 0
+        get_nodes_type = info['types'] # Get types of nodes
+        print(get_nodes_type)
+
+        for key, value in get_nodes_type.items(): # Iterate through node types and count discrete and continuous columns
+            if value == 'disc' or value == 'disc_num':
+                disc_columns += 1
+            else:
+                cont_columns += 1
+        disc_columns+=len(cat_columns)
+
+        # Based on the amount of discrete and continuous columns:
+        if cont_columns == 0:
+            self.bn = DiscreteBN() # Create Discrete Bayesian Network
+            logging.info("Using DiscreteBN")
+            self.bn.add_nodes(info)
+        elif disc_columns == 0:
+            self.bn = ContinuousBN(use_mixture=False) # Create Continuous Bayesian Network
+            logging.info("Using ContinuousBN")
+            self.bn.add_nodes(info)
+        else:
+            self.bn = HybridBN(has_logit=True, use_mixture=False) # Create Hybrid Bayesian Network
+            logging.info("Using HybridBN")
+            self.bn.add_nodes(info)
```

Comment:
`add_nodes` вообще за пределы if вынести, само выражение упростить

```suggestion

        if has_continuous and not has_discrete:
            self.bn = ContinuousBN(use_mixture=False)
            logging.info("Using ContinuousBN")
        elif has_discrete and not has_continuous:
            self.bn = DiscreteBN()
            logging.info("Using DiscreteBN")
        else:
            self.bn = HybridBN(has_logit=True, use_mixture=False)
            logging.info("Using HybridBN")
        self.bn.add_nodes(info)
```